### DISCUSSION TITLE
Implication of null values in exogenous variables with LightGBM model

### DISCUSSION THREAD

**Discussion by matsuobasho** [Nixtla/mlforecast#348] [Q&A]
I understand that missing data in exogenous variables isn't good, but in my use-case sometimes it's better to leave the missing values rather than introduce more uncertainty by filling them in with something.

I noticed that if I predict on a model with exogenous data, some of which is missing, I will get an output.  

`y_hat = fcst.predict(h=100,X_df = test)`

However, I also see the UserWarning about missing values in the exogenous variables.  What is the implication of this user warning?  From what I understand if I'm using LightGMB, then it will learn the best direction to go in a node when that variable has a missing value, and that seems acceptable.

**Comment by jmoralez** [comment]
Hey. LightGBM chooses the best direction for missing values if they are also seen during training, which isn't the case for lag features with the default settings (`dropna=True`), that's why that warning is there. If you're confident on what you're doing you can just wrap the predict call in a `warnings.catch_warnings` context manager.
